src/main/kotlin/org/rust/ide/inspections/RsSortImplTraitMembersInspection.kt

```diff
@@ -0,0 +1,58 @@
+/*
+ * Use of this source code is governed by the MIT license that can be
+ * found in the LICENSE file.
+ */
+
+package org.rust.ide.inspections
+
+import com.intellij.codeInspection.LocalQuickFix
+import com.intellij.codeInspection.ProblemDescriptor
+import com.intellij.codeInspection.ProblemsHolder
+import com.intellij.openapi.project.Project
+import org.rust.lang.core.psi.*
+import org.rust.lang.core.psi.ext.RsItemElement
+import org.rust.lang.core.psi.ext.RsTraitOrImpl
+import org.rust.lang.core.psi.ext.elementType
+import org.rust.lang.core.psi.ext.resolveToTrait
+
+class RsSortImplTraitMembersInspection : RsLocalInspectionTool() {
+    override fun buildVisitor(holder: ProblemsHolder, isOnTheFly: Boolean) = object : RsVisitor() {
+        override fun visitImplItem(implItem: RsImplItem) {
+            val traitItems = targetItems(implItem.traitRef?.resolveToTrait).takeIf { it.isNotEmpty() } ?: return
+            val implItems = targetItems(implItem).takeIf { it.isNotEmpty() } ?: return
+            if (traitItems.size != implItems.size) return
+            if (traitItems.zip(implItems).all { it.first.isSameItem(it.second) }) return
+            holder.registerProblem(
+                implItem,
+                "Different impl member order from the trait",
+                SortImplTraitMembersFix(traitItems, implItems)
+            )
+        }
+    }
+
+    private fun targetItems(item: RsTraitOrImpl?): List<RsItemElement> {
+        val children = item?.members?.children
+        return children?.mapNotNull {
+            if (it is RsFunction || it is RsConstant || it is RsTypeAlias) it as? RsItemElement else null
+        }?.takeIf { it.size == children.size } ?: emptyList()
```

Comment:
Why we need `takeIf { it.size == children.size }` here?
We return empty list if trait or impl has children that are not `RsFunction`, `RsConstant` or `RsTypeAlias` - RsInnerAttr for example
Looks like that everything will be ok if we just drop `takeIf`